src/ast/codegen_helper.cpp

```diff
@@ -0,0 +1,43 @@
+#include "codegen_helper.h"
+
+namespace bpftrace::ast {
+
+bool shouldUseWriteMapValueScratchBuffer(const AssignMapStatement &assignment)
```

Comment:
> We unnecessarily allocate the write_map_value_buf for things like @ = avg(2)

To address your point, how about having this function return `false` for aggregate

Reply:
Sounds good to me. Ah I see what I can do now...I can look at `assignment.expr` and see if its an aggregate function

Yea I'll add `is_aggregate_function` or something like that to unify logic in CodegenLLVM and ResourceAnalyser

Reply:
@ajor I took a different approach here and ended up using your suggestion of only use scratch buffer > 16 bytes. This ends up solving the aggregate problem mostly since aggregate maps use <= 16 byte values. See the PR description for more details.